stetho/src/main/java/com/facebook/stetho/server/LocalSocketHttpServer.java

```diff
@@ -260,6 +260,8 @@ public void run() {
         LogUtil.w("I/O error: %s", ex);
       } catch (HttpException ex) {
         LogUtil.w("Unrecoverable HTTP protocol violation: %s", ex);
+      } catch (OutOfMemoryError ex) {
```

Comment:
That is unfortunate.  Given this, let's catch it generically as well as specially with a comment in the general path indicating that `JSONStringer` can cause the issue as well.  That said, `JSONStringer` sucks and we really should be using a custom `JsonWriter/Reader`-based approach.